### PR TITLE
chore: update to chromium 75.0.3740.3

### PR DESCRIPTION
**Pull request by nornagon**
BREAKING CHANGE: Upgraded to Chromium 75.

Notes: Upgraded to Chromium 75.

#### FIXME:
* `IsNearDeath` is removed from upstream --> https://github.com/electron/electron/pull/17507/commits/73d71198e1c55cf314234f37dfb7af97f2f32fde

#### Requires Review:
* ~`enable_file_url_support` can no longer be enabled by embedders: https://github.com/electron/electron/pull/17507/commits/b1da14cba1b410f493400afff03b1adfc09c64ef~ Dealt with https://github.com/electron/electron/pull/17507/commits/796426a4fa58cd954fb981249d6445bdae5ef71c

### CODE REVIEW

**Comment by deepak1556** [comment]
the sdk change commit and `fix: ScopedBlockingCall now takes a location for tracing information` , should have already made it through https://github.com/electron/electron/pull/17345 , wonder why the github ui doesn't show conflicts 😕

**Comment by deepak1556** [comment]
Another important change that went in this release is that windows builds switched to chromiums libc++ https://bugs.chromium.org/p/chromium/issues/detail?id=801780 , https://docs.google.com/document/d/1VPxepzuyWbXFzv1VpkSKEn0wPvSUffpcaOLislhcwrA/edit#heading=h.4r33ra48xro . I think we should investigate the impact of this change.

Unsurprisingly the build errors are related to this :)

**Comment by deepak1556** [comment]
@jkleinsc can we enable windows debug builds for the upgrade branches ? Apart from the above change, another one that I would like ci to test is https://chromium-review.googlesource.com/c/chromium/src/+/1317069

**Comment by zcbenz** [comment]
> Another important change that went in this release is that windows builds switched to chromiums libc++

Native node modules may be affected, but I think we should be fine since the runtime library is linked dynamically in native modules.

**Comment by jkleinsc** [comment]
@deepak, I kicked off Windows debug builds

**Comment by nornagon** [comment]
> Native node modules may be affected, but I think we should be fine since the runtime library is linked dynamically in native modules.

I don't think this is true any more. See e.g. #16228, #14617

**Comment by MarshallOfSound** [comment]
> I think we should probably go with use_custom_libcxx = false on Windows.

I agree, I'm getting a Windows VM set up at the moment so I'll give this a spin shortly :+1:

**Comment by KoenLav** [comment]
@nornagon @marshallofsound please forgive my ignorance, but why stick with the "non-standard"/community supported route?

**Comment by nornagon** [comment]
@KoenLav because of node native modules.

node native modules don't expect to be built against libc++ on windows & such may not compile. Additionally, we'd have to ship libc++'s headers with the electron header bundle and upstream changes to node-gyp and cmake-js to link against libc++ instead of msstl.

it's a complicated situation with no great answers. sticking with msstl is the easiest option in the short term, but we may have to figure out something else in the long run.

File: atom/browser/atom_browser_client.cc [hunk 1]
```diff
@@ -830,7 +830,11 @@ bool AtomBrowserClient::HandleExternalProtocol(
     ui::PageTransition page_transition,
     bool has_user_gesture,
     const std::string& method,
-    const net::HttpRequestHeaders& headers) {
+    const net::HttpRequestHeaders& headers,
+    network::mojom::URLLoaderFactoryRequest* factory_request,
+    // clang-format off
+    network::mojom::URLLoaderFactory*& out_factory) {  // NOLINT
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply atom/browser/atom_browser_client.cc]: Yup 😢 Need to disable both

File: patches/common/chromium/command-ismediakey.patch [hunk 2]
```diff
@@ -58,14 +58,6 @@ index f5d0ac4ba826ee4f023224ce90c6d8b6614c2912..380682029d20cab485bb8b52bd5c7d9c
    ui::Accelerator accelerator(
        ui::KeyboardCodeForWindowsKeyCode(key_code), modifiers);
  
-@@ -98,6 +100,7 @@ bool GlobalShortcutListenerWin::RegisterAcceleratorImpl(
-   modifiers |= accelerator.IsShiftDown() ? MOD_SHIFT : 0;
-   modifiers |= accelerator.IsCtrlDown() ? MOD_CONTROL : 0;
-   modifiers |= accelerator.IsAltDown() ? MOD_ALT : 0;
-+  modifiers |= accelerator.IsCmdDown() ? MOD_WIN : 0;
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/common/chromium/command-ismediakey.patch]: Hm, maybe a bad rebase, I'll look at this tomorrow morning

File: patches/common/chromium/fix_retain_compatibility_with_msvc.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/common/chromium/fix_retain_compatibility_with_msvc.patch]: No, I'm working on it though.  👍

File: atom/browser/notifications/platform_notification_service.cc [hunk 4]
```diff
@@ -134,4 +134,13 @@ void PlatformNotificationService::RecordNotificationUkmEvent(
     content::BrowserContext* browser_context,
     const content::NotificationDatabaseData& data) {}
 
+void PlatformNotificationService::ScheduleTrigger(
+    content::BrowserContext* browser_context,
+    base::Time timestamp) {}
```
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply atom/browser/notifications/platform_notification_service.cc]: First pass was it was a lot of effort to support (it uses a lot of //chrome code) the shell impl is empty iirc, we might be able to though

File: atom/browser/atom_browser_client.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (deepak1556) [reply atom/browser/atom_browser_client.cc]: why disable both ? Can we keep one over the other (prefer clang-format rules where possible). Upstream feels the same way about these stuff https://groups.google.com/a/chromium.org/d/msg/chromium-dev/I2fjO9pSVDU/VdKONuzyEgAJ

[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply atom/browser/atom_browser_client.cc]: @deepak1556 So the issue here is the lint fails, to disable the lint we need to use `// NOLINT`.  But then the line is too long and `clang-format` tries to fix it by putting `// NOLINT` on the next line

[the remark this answers is not in the document]
Comment (deepak1556) [reply atom/browser/atom_browser_client.cc]: Ah okay, thats sad.

File: patches/common/chromium/fix_retain_compatibility_with_msvc.patch (hunk 3 above)
[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/common/chromium/fix_retain_compatibility_with_msvc.patch]: So turns out this was fixed 2 days in Chrome master 👍 

They've disabled the custom libc++ again as it broke Visual Studio debugging.

https://chromium-review.googlesource.com/c/chromium/src/+/1546066

I'll monitor to see if they add it back 👍

[the remark this answers is not in the document]
Comment (nornagon) [reply patches/common/chromium/fix_retain_compatibility_with_msvc.patch]: > if they add it back

they definitely will; it looks like that revert was just temporary while they work on a vis tool for the VS debugger.

[the remark this answers is not in the document]
Comment (MarshallOfSound) [reply patches/common/chromium/fix_retain_compatibility_with_msvc.patch]: s/if/when 👍

File: atom/browser/ui/inspectable_web_contents_impl.cc [hunk 5]
```diff
@@ -738,7 +739,8 @@ void InspectableWebContentsImpl::DispatchProtocolMessage(
   if (message.length() < kMaxMessageChunkSize) {
     base::string16 javascript =
         base::UTF8ToUTF16("DevToolsAPI.dispatchMessage(" + message + ");");
```
Comment (nornagon) [inline atom/browser/ui/inspectable_web_contents_impl.cc]: lol really, we don't escape this string at all?

File: build/zip.py [hunk 6]
```diff
@@ -52,7 +52,7 @@ def main(argv):
   if sys.platform == 'darwin':
     execute(['zip', '-r', '-y', dist_zip] + list(dist_files))
   else:
-    with zipfile.ZipFile(dist_zip, 'w', zipfile.ZIP_DEFLATED) as z:
+    with zipfile.ZipFile(dist_zip, 'w', zipfile.ZIP_DEFLATED, True) as z:
```
Comment (nornagon) [inline build/zip.py]:
```suggestion
    with zipfile.ZipFile(dist_zip, 'w', zipfile.ZIP_DEFLATED, allowZip64=True) as z:
```

File: patches/common/chromium/disable_custom_libcxx_on_windows.patch [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+index 5d45ff18455275e2ab416030fb01b103c7a630cf..39774eb51b112e02a9aa9c5288e8b52e33fdcb17 100644
+--- a/build/config/c++/c++.gni
++++ b/build/config/c++/c++.gni
+@@ -13,7 +13,8 @@ declare_args() {
+   # TODO(https://crbug.com/942939): Re-enable libc++ on Windows with libfuzzer.
+   use_custom_libcxx =
+       is_fuchsia || is_android || is_mac ||
+-      (is_win && is_clang && !use_libfuzzer) ||
++      # Do not use custom libcxx on windows
++      # (is_win && is_clang && !use_libfuzzer) ||
```
Comment (nornagon) [inline patches/common/chromium/disable_custom_libcxx_on_windows.patch]: i'd recommend adding `&& !is_electron` here rather than commenting out.

**Pull request merged**